patches/proxy-uri.diff

```diff
@@ -125,7 +125,11 @@ Index: code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
 +
 +			if (localhostMatch && resolvedUri.authority !== location.host) {
 +				if (config.productConfiguration && config.productConfiguration.proxyEndpointTemplate) {
-+					resolvedUri = URI.parse(new URL(config.productConfiguration.proxyEndpointTemplate.replace('{{port}}', localhostMatch.port.toString()), window.location.href).toString())
++					const renderedTemplate = config.productConfiguration.proxyEndpointTemplate
++												.replace('{{port}}', localhostMatch.port.toString())
++												.replace('{{host}}', window.location.hostname)
```

Comment:
I don't think that would be a good idea. That is the whole reason I use window.location.hostname, that the port is NOT included in it. Otherwise the `{{host}}` part is only allowed at the end and not for example `code-{{port}}-{{host}}.mytunneldomain.tld`. But I assumed, that the `URI.parse(` would do the right thing and join them together.

But in the end: I think it is better this way. The user should provide his own VSCODE_PROXY_URI definition with the port present e.g. `http://{{port}}.{{host}}:8080`. Otherwise we will assume something on that URI which might not be true.

Before this PR the automatism of setting VSCODE_PROXY_URI if not present did not work correctly anyway. It was not prefixed by the protocol or the placeholder `//`, which let to something like this for e.g. proxy-domain `domain.tld`:
`http://localhost:8080/8000.domain.tld`. I added a fix for that as well by prefixing the proxy-domain by `//`.

Reply:
On second thought, we could support post: host and hostname. What do you think?

Reply:
> Otherwise the {{host}} part is only allowed at the end

I think `{{host}}` really only makes sense at the end anyway.  If you did `{{host}}.tunnel.tld` then the ports panel would say `https://tunnel.tld.tunnel.tld` which is probably also wrong anyway?

```javascript
console.log(URI.parse("https://tunnel.tld.tunnel.tld", "https://tunnel.tld")) // Outputs https://tunnel.tld.tunnel.tld/
```

> Before this PR the automatism of setting VSCODE_PROXY_URI if not present did not work correctly anyway

Good point!  Nice catch on this bug.

> The user should provide his own VSCODE_PROXY_URI definition with the port present

My understanding of `{{host}}` was that it was for cases where you had no control over the host or port but that it was guaranteed to be the same as code-server's, so if I have that right then we want to keep the port.

Like if ngrok gives you `https://*.domain.tld:8080` then you want to keep that 8080 (I think ngrok only gives you 80/433, but just as an example).  I might have misunderstood though.

Are there scenarios where the port would be different?  If so then an option makes sense to me.  We could also add `hostname` later if we are not aware of any current use cases.  But since code-server and the proxy are the same process bound to the same port I would assume the port would always be the same as well.

Reply:
that would work if you have e.g. `https://mycode.domain.tld`  and `https://{{host}}-{{port}}.tunnelhost.tld`, but I guess the use case is not very high, so I think you are right, this feature will most likely only be used at the end for the proxy-domain. So I change it to `window.location.host`